lib/ui/painting/immutable_buffer.cc

```diff
@@ -53,6 +58,49 @@ size_t ImmutableBuffer::GetAllocationSize() const {
   return sizeof(ImmutableBuffer) + data_->size();
 }
 
+void ImmutableBuffer::fromAsset(Dart_NativeArguments args) {
+  UIDartState::ThrowIfUIOperationsProhibited();
+  Dart_Handle callback_handle = Dart_GetNativeArgument(args, 2);
+  if (!Dart_IsClosure(callback_handle)) {
+    Dart_SetReturnValue(args, tonic::ToDart("Callback must be a function"));
+    return;
+  }
+  Dart_Handle asset_name_handle = Dart_GetNativeArgument(args, 0);
+  uint8_t* chars = nullptr;
+  intptr_t asset_length = 0;
+  Dart_Handle result =
+      Dart_StringToUTF8(asset_name_handle, &chars, &asset_length);
+  if (Dart_IsError(result)) {
+    Dart_PropagateError(result);
+    return;
+  }
+  Dart_Handle immutable_buffer = Dart_GetNativeArgument(args, 1);
+
+  std::string asset_name = std::string{reinterpret_cast<const char*>(chars),
+                                       static_cast<size_t>(asset_length)};
+
+  std::shared_ptr<AssetManager> asset_manager = UIDartState::Current()
+                                                    ->platform_configuration()
+                                                    ->client()
+                                                    ->GetAssetManager();
+  std::unique_ptr<fml::Mapping> data = asset_manager->GetAsMapping(asset_name);
+
+  if (data == nullptr) {
+    return;
+  }
+  const void* bytes = static_cast<const void*>(data->GetMapping());
+  void* peer = reinterpret_cast<void*>(data.release());
+  auto size = data->GetSize();
+  SkData::ReleaseProc proc = [](const void* ptr, void* context) {
+    // TODO: what do
+  };
+
+  auto sk_data = SkData::MakeWithProc(bytes, size, proc, peer);
```

Comment:
We might want to just store the mapping object in our heap which should keep it alive, and then `reset` it in the release proc. I think that matches the semantics of the fml::Mapping, but @jason-simmons or @chinmaygarde would know for sure.

Reply:
The `peer` will be passed to the `ReleaseProc` as the `context` parameter.  The `ReleaseProc` can cast it back to an `fml::Mapping*` and delete it.

Reply:
Will that be enough to keep the Mapping pointer alive?

Reply:
The mapping pointer should remain valid as long as the `fml::Mapping` and the `AssetManager` that created it have not been deleted.